app/request_handlers/handlers/lobby.js

```diff
@@ -43,21 +60,62 @@ const lobbyJoinGame = (req, res) => {
     return res.send({ role });
 }
 
+const lobbyLeave = (req, res) => { // wasn't sure if lobbyId should be specified by client, leaving it as option for now
+	const body = req.body;
+	const username = body.username;
+	let lobbyId = 0;
+	if (body.lobbyId === undefined) {
+		if (Object.keys(userJoined).includes(String(username))) {
+			lobbyId = userJoined[username];
+		}
+		else {
+			throw Error("User is not currently in a lobby.");
+		}
+	}
+	else {
+		lobbyId = body.lobbyId;
+	}
+	console.log("POST /lobby/leave", body);
+	
+	leaveLobby(lobbyId, username);
+	delete userJoined[username];
+	refreshLobbyPage();
```

Comment:
Could users ever leave a lobby though means other than sending a Post request to /lobby/leave? For example, maybe the server would kick players out of a lobby if they log out.
Let's move this lobby-leaving logic into a new function, and have this Post request call that, rather than cluttering this request handler